inspirehep/modules/tools/utils.py

```diff
@@ -0,0 +1,151 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2016 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this licence, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""Utility functions for various tools."""
+
+from __future__ import absolute_import, print_function
+
+import re
+
+import six
+from nameparser import HumanName
+
+
+def authorlist(text):
+    """Call the relevant formatter depending on the existence of affiliations.
+
+    Input can be in two different formats:
+        F. Lastname, F.M. Otherlastname
+    or
+        F. Lastname1, F.M. Otherlastname1,2
+        1 CERN
+        2 Otheraffiliation
+    """
+    if not text:
+        return ''
+    if not isinstance(text, six.text_type):
+        # FIXME: how should I handle unicode?
+        text = text.decode('utf-8')
+
+    # Assume that if there are numbers in author name, they are affilation ids.
+    # Also assume that if one author has an affiliation, everyone else has too.
+    # FIXME: can we make these assumptions?
```

Comment:
Is this really a FIXME? This is a question about requirements, which I believe only you and the curators know...